optimade/grammar/v0.10.0.lark

```diff
@@ -0,0 +1,90 @@
+// optimade v0.10.0 grammar spec in lark grammar format
```

Comment:
With the changes we should consider renaming this grammar to v0.10.1

Reply:
This is again a question about versions and I'm not sure worth it. Should we keep `v0.10.0.lark` as well? If we start doing this we must align the versions of the release, the PyPI and the grammar manually every time...

Reply:
I don't think we need to do this, see my comment [here](https://github.com/Materials-Consortia/optimade-python-tools/issues/72#issuecomment-553833440).